### DISCUSSION TITLE
Domínios Ricos: como criar os objetos das entidades a partir de dados do banco?

### DISCUSSION THREAD

**Discussion by henriqueweege** [balta-io/forum#380] [Tira Dúvidas] [answered]
Olá! Tudo bem?
Então, estou refatorando o exercício do curso Dominando Injeção de Dependência para aplicar alguns conceitos de domínio rico e expandir o projeto para utilizar CQRS/Mediator, e implementar algumas coisas de DevOps, como pipeline de CI/CD etc.

Pois bem, minha dúvida é a seguinte: uma vez que vou precisar de uma model para representação da entidade para persistir em algum banco de dados, e existem propriedades da entidade que são geradas dentro dela, como por exemplo o Id, como proceder no caso de criar instanciar um objeto do tipo "entidade" a partir de uma model com informações persistidas no banco? Pensei em por o parâmetro como opcional no construtor, ou ter uma flag para diferenciar o que é um objeto novo do que é um objeto ja persistido, mas ambas soluções parecem deixar brecha para mal uso. 

Outro exemplo é no caso de um Put, onde eu receberia um command com todas as informações da entidade, geraria um objeto do tipo "entidade" para passar pelas validações e a partir desse objeto seria instanciado um novo objeto do tipo "model" para enviar ao banco. Nesse caso, também me parece complicado passar algum parâmetro com essas características do command para a entidade.

Poderia me ajudar?

Muito obrigado desde já pela atenção.
Henrique

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @henriqueweege, como vai? Quando as entidades são criadas a partir do banco, normalmente os ORMs ou MicroORMs utilizam o que chamamos de proxy. Eles fazem uma cópia da classe e preenchem todas as propriedades.

A geração do ID como você comentou, só acontece quando usamos o `new` para criar uma nova instância do objeto.

**Reply by henriqueweege** [reply]
Bom dia!
Por aqui tudo bem, e por ai?
Aah, eu não sabia disso haha obrigadão, Balta!!
